src/engine/V3Eval.v3

```diff
@@ -823,6 +861,47 @@ def do_v_v_x16(a: (u64, u64), f: (u8) -> u8) -> (u64, u64) { // Performs a 16-la
 	}
 	return (low, high);
 }
+// Binary v128 ops, in the form of (v, i32) -> v
+def do_vi_v_x16(a: (u64, u64), b: i32, f: (u8, u8) -> u8) -> (u64, u64) {
+	var low: u64 = 0;
+	var high: u64 = 0;
+	for (shift: byte = 0; shift < 64; shift += 8) {
+		var r_a = u8.view(a.0 >> shift);
+		var res = f(r_a, u8.view(b));
+		low |= (u64.view(res) << shift);
+		
+		r_a = u8.view(a.1 >> shift);
+		res = f(r_a, u8.view(b));
+		high |= (u64.view(res) << shift);
+	}
+	return (low, high);
+}
+def do_vi_v_x8(a: (u64, u64), b: i32, f: (u16, u16) -> u16) -> (u64, u64) {
+	var low: u64 = 0;
+	var high: u64 = 0;
+	for (shift: byte = 0; shift < 64; shift += 16) {
```

Comment:
This will be a little more efficient if you just loop with a regular `int` induction variable and then do a `u6.view` once for each iteration.

Reply:
Updated. Also fix places where code is sub-optimal.